src/iso15118/d20/state/supported_app_protocol.cpp

```diff
@@ -22,10 +22,16 @@ handle_request(const message_20::SupportedAppProtocolRequest& req) {
         if (protocol.protocol_namespace.compare("urn:iso:std:iso:15118:-20:DC") == 0) {
             res.schema_id = protocol.schema_id;
             selected_protocol = "ISO15118-20:DC";
-            return {response_with_code(
-                        res, message_20::SupportedAppProtocolResponse::ResponseCode::OK_SuccessfulNegotiation),
-                    selected_protocol};
         }
+
+        if (protocol.protocol_namespace.compare("urn:iso:std:iso:15118:-20:AC") == 0) {
+            res.schema_id = protocol.schema_id;
+            selected_protocol = "ISO15118-20:AC";
+        }
+
+        return {
+            response_with_code(res, message_20::SupportedAppProtocolResponse::ResponseCode::OK_SuccessfulNegotiation),
+            selected_protocol};
```

Comment:
Should this return statement be wrapped in a `if (selected_protocol.has_value())`?
Currently if the protocol list from the request has a not-supported protocol first or if the list does not include D20-AC/DC at all then it would respond `OK_SuccessfulNegotiation` nonetheless

Reply:
With this new commit the problem should be fixed